File: iOS/Advanced_Demo/Advanced/Vender/ToastComponents/ToastView.m
//
//  ToastView.m
//  veRTC_Demo
//
//

#import "ToastView.h"
#import "Masonry.h"

#define SCREEN_WIDTH  ([UIScreen mainScreen].bounds.size.width)
#define SCREEN_HEIGHT ([UIScreen mainScreen].bounds.size.height)
#define XH_1PX_WIDTH (1 / [UIScreen mainScreen].scale)

@interface ToastView ()

@property (nonatomic, strong) UIView *bgView;

@end


@implementation ToastView

- (instancetype)initWithMeeage:(NSString *)message {
    self = [super init];
    if (self) {
        CGFloat minScreen = MIN(SCREEN_WIDTH, SCREEN_HEIGHT);
        CGFloat scale = (minScreen / 375);
        self.bgView = [[UIView alloc] init];
        self.bgView.backgroundColor = [UIColor colorWithWhite:0.1 alpha:0.8];
        self.bgView.layer.cornerRadius = 4;
        self.bgView.layer.masksToBounds = YES;
        
        UILabel *titleLabel = [[UILabel alloc] init];
        titleLabel.numberOfLines = 0;
        titleLabel.textColor = [UIColor whiteColor];
        titleLabel.textAlignment = NSTextAlignmentCenter;
        titleLabel.font = [UIFont systemFontOfSize:16.0 * scale weight:UIFontWeightRegular];
        titleLabel.text = message;
        
        [self addSubview:self.bgView];
        [self.bgView addSubview:titleLabel];
        
        [titleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
            make.center.equalTo(self.bgView);
            make.width.mas_lessThanOrEqualTo(SCREEN_WIDTH - 24 * 2);
        }];
        
        [self.bgView mas_makeConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(titleLabel).offset(-12);
            make.right.equalTo(titleLabel).offset(12);
            make.top.equalTo(titleLabel).offset(-12);;
            make.bottom.equalTo(titleLabel).offset(12);;
        }];
        
        [self mas_updateConstraints:^(MASConstraintMaker *make) {
            make.size.equalTo(self.bgView);
        }];
    }
    return self;
}

@end

File: iOS/ApiExample/ApiExample/Common/ToastView.swift
//
//  ToastView.swift
//  ApiExample
//
//  Created by bytedance on 2023/11/3.
//  Copyright © 2021 bytedance. All rights reserved.
//

import UIKit
import SnapKit

class ToastView: UIView {

    private var bgView = UIView()

    init(message: String) {
        super.init(frame: .zero)
        
        let minScreen = min(UIScreen.main.bounds.width, UIScreen.main.bounds.height)
        let scale = (minScreen / 375)
        bgView = UIView()
        bgView.backgroundColor = UIColor(white: 0.1, alpha: 0.8)
        bgView.layer.cornerRadius = 4
        bgView.layer.masksToBounds = true
        
        let titleLabel = UILabel()
        titleLabel.numberOfLines = 0
        titleLabel.textColor = .white
        titleLabel.textAlignment = .center
        titleLabel.font = UIFont.systemFont(ofSize: 16.0 * scale, weight: .regular)
        titleLabel.text = message
        
        self.addSubview(bgView)
        bgView.addSubview(titleLabel)

        titleLabel.snp.makeConstraints { make in
            make.center.equalTo(bgView)
            make.width.lessThanOrEqualTo(UIScreen.main.bounds.width - 24 * 2)
        }

        bgView.snp.makeConstraints { make in
            make.left.equalTo(titleLabel).offset(-12)
            make.right.equalTo(titleLabel).offset(12)
            make.top.equalTo(titleLabel).offset(-12)
            make.bottom.equalTo(titleLabel).offset(12)
        }

        self.snp.updateConstraints { make in
            make.size.equalTo(bgView)
        }
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
